### PR TITLE
Eslint - Add scoping to switch statement declarations

### PR DESCRIPTION
**Pull request by zacnomore**
To help address #200 

This adds blocks to each case to ensure scope is intuitive
More details: https://eslint.org/docs/rules/no-case-declarations

### CODE REVIEW

**Comment by iHiD** [comment]
@SleeplessByte @ErikSchierboom @kntsoriano Could one of you take a look pls.

File: app/javascript/components/misc/CodeMirror/languageCompartment.ts [hunk 1]
```diff
@@ -7,177 +7,227 @@ export const loadLanguageCompartment = async (
...
       return compartment.of(java())
-    case 'cfml':
+    }
+    case 'cfml': {
       const { javascript: cfml } = await import('@codemirror/lang-javascript')
       return compartment.of(cfml())
+    }
     case 'javascript':
-    case 'typescript':
+    case 'typescript': {
```
Comment (ErikSchierboom) [inline app/javascript/components/misc/CodeMirror/languageCompartment.ts:23]: The documentation's "When not to use" section says:

> If you depend on fall through behavior and want access to bindings introduced in the case block.

Should we copy the code to not have fall through behavior then?

Comment (SleeplessByte) [reply app/javascript/components/misc/CodeMirror/languageCompartment.ts:23]: No, this works.

```typescript
switch (language) {
  case 'javascript':
  case 'typescript': { 
    // runs on both js and ts
    console.log(language);
  } 
}

// => 'javascript' is logged
```

**Review by SleeplessByte — APPROVED** [review]

**Review by kntsoriano — APPROVED** [review]

**Review by ErikSchierboom — APPROVED** [review]
Thanks!

**Pull request merged**